src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -900,6 +942,25 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
             BoundExpression boundOperand = boundConversion.Operand;
             if (boundConversion.ConversionKind == CSharp.ConversionKind.MethodGroup)
             {
+                if (boundConversion is { Type: FunctionPointerTypeSymbol { Signature: var signature }, Syntax: PrefixUnaryExpressionSyntax addressOfExpression })
```

Comment:
>PrefixUnaryExpressionSyntax [](start = 111, length = 27)

How do we know this is an address of? Besides, is it really important what syntax we have here? We have a bound node, semantics of bound nodes doesn't depend on associated syntax node. The translation should be based on semantics. Is this a method group conversion to a function pointer? Then it is not a delegate creation regardless of the syntax.